plugins/aws/cli_provisioner.go

```diff
@@ -25,22 +26,36 @@ func (p awsCLIProvisioner) Provision(ctx context.Context, in sdk.ProvisionInput,
 
 func stripAndReturnProfileFlag(args *[]string) (string, error) {
 	for i, arg := range *args {
+		// if `--profile` is used after `--`, it should not be interpreted as a flag
+		if arg == "--" {
+			break
+		}
+
 		if arg == "--profile" {
 			if i+1 == len(*args) {
 				return "", fmt.Errorf("--profile flag was specified without a value")
 
 			}
 			profile := (*args)[i+1]
 			// Remove --profile flag so aws cli doesn't attempt to assume role by itself
-			*args = removeProfileFlagFromArgs(i, *args)
+			*args = removeArgSequenceFromArgList(i, i+1, *args)
+			return profile, nil
+		} else if strings.Contains(arg, "--profile=") {
+			split := strings.Split(arg, "=")
+			if len(split) != 2 {
+				return "", fmt.Errorf("--profile flag was specified without a value")
+			}
+			profile := split[1]
+			// Remove --profile flag so aws cli doesn't attempt to assume role by itself
```

Comment:
Mostly as an FYI: Go also has a specific function for this:
```go
profile := strings.TrimPrefix(arg, "--profile=")
if profile == "" {
    return "", fmt.Errorf("--profile flag was specified without a value")
}
```

Feel free to leave it as-is though, the current implementation works just fine :)